pkg/driver/nri_hooks.go

```diff
@@ -247,18 +251,27 @@ func (np *NetworkDriver) StopPodSandbox(ctx context.Context, pod *api.PodSandbox
 		// some version of containerd does not send the network namespace information on this hook so
 		// we workaround it using the local copy we have in the db to associate interfaces with Pods via
 		// the network namespace id.
-		ns = np.netdb.GetPodNamespace(podKey(pod))
+		ns = np.netdb.GetPodNetNs(podKey(pod))
 		if ns == "" {
 			klog.Infof("StopPodSandbox pod %s/%s using host network ... skipping", pod.Namespace, pod.Name)
 			return nil
 		}
 	}
 
 	for deviceName, config := range podConfig {
-		ifName := names.GetOriginalName(deviceName)
-
-		if err := nsDetachNetdev(ns, config.Network.Interface.Name, ifName); err != nil {
-			klog.Infof("fail to return network device %s : %v", deviceName, err)
+		// TODO(gauravkghildiyal): Interface name here will in most cases be
+		// empty which means we're mostly relying on the fallback of the kernel
+		// having to do this cleanup. That is hopefully not the worst thing
+		// which could happen, but there are easy ways to improve this
+		// behaviour, like detaching the specific network interface which has
+		// the the given PCI address (i.e. deviceName is the normalized PCI address).
+		ifName, err := np.netdb.GetNetInterfaceName(deviceName)
+		if err == nil {
+			if err := nsDetachNetdev(ns, config.Network.Interface.Name, ifName); err != nil {
+				klog.Infof("fail to return network device %s : %v", deviceName, err)
+			}
```

Comment:
I see my confusion, I knew we were moving the interface back, but I didn't remember where.
I'm not sure now when this hooks run, but if nodeUnprepareResource runs before this , can't we just do it there?